scripts/ci/check_compliance.py

```diff
@@ -253,6 +256,38 @@ def required_false_check(self, dts_binding):
                         line_number, col=None,
                         desc="'required: false' is redundant, please remove")
 
+    def compatible_format(self, dts_binding):
+        with open(dts_binding) as file:
+            binding = yaml.load(file, Loader=yaml.SafeLoader)
+            compatible = binding.get("compatible", None)
+
+            if not compatible:
+                return
+
+            pattern = re.compile(r"[a-z0-9+,\-._]*")
+            if not pattern.fullmatch(compatible):
+                self.failure(f"""\
+Invalid characters in {dts_binding} compatible: "{compatible}".
+Use only lowercase letters, numbers and '+' ',' '-' '.' '_'.""")
+
+    def compatible_properties(self, dts_binding):
+        with open(dts_binding) as file:
+            binding = yaml.load(file, Loader=yaml.SafeLoader)
+
+            property_names = []
+            property_names.extend(binding.get("properties", {}))
+
+            child_binding = binding.get("child-binding", None)
+            if child_binding:
+                property_names.extend(child_binding.get("properties", {}))
```

Comment:
note: we can have child-binding at arbitrary levels (forming e.g. grandchild binding)

Reply:
Good point, let's iterate on those as well.